tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+        
+            // Expected return value = 2 + 3 + 4 + 6 + 7 + 8 + 11 + 12 + 13 = 93
+            float sum = cdt.vecs.a.X + cdt.vecs.a.Y + cdt.vecs.a.Z 
+                + cdt.vecs.b.X + cdt.vecs.b.Y + cdt.vecs.b.Z 
+                + cdt.v3.X + cdt.v3.Y + cdt.v3.Z;
+            if ((sum != 93) || (cdt.iv != 100) || (cdt.str.ToString() != "ret_string") )
+            {
+                Console.WriteLine("PInvoke Vector3InStruct test failed");
+                return false;            
+            }
+        }        
+
+        Console.WriteLine("All PInvoke testcases passed");
+        return true;
+    }    
+}
+
+class RPInvokeTest 
+{
+    public delegate void CallBackDelegate_RPInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2);
+        
+    public delegate Vector3 CallBackDelegate_RPInvoke_Vector3Ret();
+    
+    public delegate void CallBackDelegate_RPInvoke_Vector3Array(
+        [In, MarshalAs(UnmanagedType.LPArray, SizeParamIndex=1)] Vector3[] v, 
+        int size); 
+        
+    public delegate void CallBackDelegate_RPInvoke_Vector3InStruct(
+        DT v);        
+        
+    public delegate void CallBackDelegate_RPInvoke_Vector3InComplexStruct(
+        ref ComplexDT v);    
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3Arg(
+        CallBackDelegate_RPInvoke_Vector3Arg callBack);
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern bool nativeCall_RPInvoke_Vector3Ret(
+        CallBackDelegate_RPInvoke_Vector3Ret callBack);        
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3Array(
+        CallBackDelegate_RPInvoke_Vector3Array callBack, 
+        int v);
+           
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_RPInvoke_Vector3InStruct(
+        CallBackDelegate_RPInvoke_Vector3InStruct callBack, 
+        int v);
+        
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]        
+    public static extern bool nativeCall_RPInvoke_Vector3InComplexStruct(
+        CallBackDelegate_RPInvoke_Vector3InComplexStruct callBack);
+    
+    static bool result = false;
+   
+    static void callBack_RPInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2)
+    {
+        Vector3 tmp = new Vector3(2, 2, 2);
+        
+        // sum = (1, 2, 3) dot (2, 2, 2) = 12
+        float sum0 = Vector3.Dot(v1, tmp);   
+        // sum = (10, 20, 30) dot (2, 2, 2) = 20 + 40 + 60 = 120
+        float sum1 = Vector3.Dot(v2, tmp);        
```

Comment:
The above method signature on Linux would mean that both v1 and v2 will  be in registers (unix allows 6 reg params).  Please modify it to have some addition args before v2 so that v2 is passed on stack.